human_cancer_analysis/nbconverted/2_identify_generic_genes_pathways.py

```diff
@@ -0,0 +1,778 @@
+    os.path.join(base_dir, "configs", "config_human_cancer.tsv")
+)
+
+params = utils.read_config(config_filename)
+
+
+# In[3]:
+
+
+# Load params
+local_dir = params["local_dir"]
+dataset_name = params['dataset_name']
+NN_architecture = params['NN_architecture']
+num_runs = params['num_simulated']
+project_id = params['project_id']
+metadata_col_id = params['metadata_colname']
+processed_template_filename = params['processed_template_filename']
+normalized_compendium_filename = params['normalized_compendium_filename']
+scaler_filename = params['scaler_filename']
+col_to_rank_genes = params['rank_genes_by']
+col_to_rank_pathways = params['rank_pathways_by']
+compare_genes = params['compare_genes']
+statistic = params['gsea_statistic']
+
+# Load metadata file with grouping assignments for samples
+sample_id_metadata_filename = os.path.join(
+    base_dir,
+    dataset_name,
+    "data",
+    "metadata",
+    f"{project_id}_process_samples.tsv"
+)
+
+# Load pickled file
+with open(scaler_filename, "rb") as scaler_fh:
+    scaler = pickle.load(scaler_fh)
+
+
+# In[4]:
+
+
+# Output files
+gene_summary_filename = os.path.join(
+    base_dir, 
+    dataset_name, 
+    f"generic_gene_summary_{project_id}.tsv"
+)
+
+pathway_summary_filename = os.path.join(
+    base_dir, 
+    dataset_name, 
+    f"generic_pathway_summary_{project_id}.tsv"
+)
+
+
+# ## Need to customize code from ponyo
+# 
+# The current simulation-related function in ponyo, `get_sample_ids` assumes that the user is using one of two different metadata files (one associated with the pseudomonas compendium and another associated with recount2). The compendium dataset we are using here has a slightly different format for their metadata file. 
+# 
+# Here we are temporarily writing our own function customized for this Powers et. al. dataset. But we will be updating ponyo to allow for different metadata files in the future. Issue in ponyo is [here](https://github.com/greenelab/ponyo/issues/18) 
+
+# In[5]:
+
+
+def get_sample_ids(experiment_id, dataset_name, sample_id_colname):
+    """
+    Returns sample ids (found in gene expression df) associated with
+    a given list of experiment ids (found in the metadata)
+
+    Arguments
+    ----------
+    experiment_ids_file: str
+        File containing all cleaned experiment ids
+
+    dataset_name: str
+        Name for analysis directory. Either "Human" or "Pseudomonas"
+
+    sample_id_colname: str
+        Column header that contains sample id that maps expression data
+        and metadata
+
+    """
+    base_dir = os.path.abspath(os.path.join(os.getcwd(), "../"))
+
+    # Read in metadata
+    metadata = pd.read_csv(metadata_filename, header=0)
+    metadata.set_index("gse", inplace=True)
+
+    selected_metadata = metadata.loc[experiment_id]
+    sample_ids = list(selected_metadata[sample_id_colname])
+
+    return sample_ids
+
+
+# In[6]:
+
+
+import glob
+from keras.models import load_model
+
+def shift_template_experiment(
```

Comment:
Is this the same as the function in `ponyo`, except with `metadata_filename` as an argument? Might be good to make a note of that, so you remember in the future.

Reply:
Yes I hackily changed it for this one case since it would take a bit to update ponyo. I should update the name of the function to not confuse myself, good point